addons/modules/functions/fnc_bi_moduleCurator.sqf

```diff
@@ -0,0 +1,397 @@
+#include "script_component.hpp"
+/*
+ * Author: Bohemia Interactive
+ * Module function for initializing the Zeus Game Master module.
+ * Edited to allow for control over ascension messages and camera eagle,
+ * fix double attributes dialog bug, fix respawn on start with forced interface bug,
+ * and fix loss of curator assignment when the module's locality is transferred.
+ *
+ * Arguments:
+ * 0: Logic <OBJECT>
+ * 1: Units <ARRAY>
+ * 2: Activated <BOOL>
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [LOGIC, [], true] call zen_modules_fnc_bi_moduleCurator
+ *
+ * Public: No
+ */
+
+params ["_logic", "_units", "_activated"];
+
+if (_activated) then {
+    // BI's module framework is bugged and calls module function twice on clients after mission start
+    if (_logic getVariable [QGVAR(initialized), false]) exitWith {};
+
+    // Mark the module as initialized on this machine, fixes double attributes dialog bug
+    _logic setVariable [QGVAR(initialized), true];
+
+    // Exit if the module is not created on the server
+    if (!isServer && {local _logic} && {isNull getAssignedCuratorUnit _logic}) exitwith {
+        [format ["%1 is trying to create a Curator logic ModuleCurator_F.", profileName], "BIS_fnc_error", false] call BIS_fnc_MP;
+        deleteVehicle _logic;
+    };
+
+    // Determine the curator module's owner
+    private _ownerVar = _logic getVariable ["owner", ""];
+    private _isUID = parseNumber _ownerVar > 0;
+
+    if (cheatsEnabled) then {
+        private _ownerVarArray = toArray _ownerVar;
+        _ownerVarArray resize 3;
+
+        if (toString _ownerVarArray == "DEV") then {
+            _isUID = true;
+        };
+    };
+
+    if (_ownerVar == "" && {!isMultiplayer}) then {
+        ["Curator owner not defined, player used instead in singleplayer."] call BIS_fnc_error;
+        _ownerVar = player call BIS_fnc_objectVar;
+    };
+
+    if (_isUID && {!isMultiplayer}) then {
+        _ownerVar = player call BIS_fnc_objectVar;
+    };
+
+    private _isAdmin = _ownerVar == "#adminLogged" || {_ownerVar == "#adminVoted"};
+
+    // Wipe out the variable so clients can't access it
+    _logic setVariable ["owner", nil];
+
+    if (isServer) then {
+        // Prepare admin variable
+        private _adminVar = "";
+
+        if (_isAdmin) then {
+            private _letters = ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", "k", "l", "m", "n", "o", "p", "q", "r", "s", "t", "u", "v", "w", "x", "y", "z"];
```

Comment:
Maybe to a define? To keep it a bit cleaner.

Reply:
Don't think it will make a difference, this is BI code anyways.